### PR TITLE
Meta language: Modified parser to handle SGVector, SGMatrix accessing

### PR DESCRIPTION
**Pull request by sorig**
WIP

### CODE REVIEW

File: examples/meta/.gitignore [hunk 1]
```diff
@@ -1,3 +0,0 @@
-generator/parsetab.py
```
Comment (karlnapf) [inline examples/meta/.gitignore]: +1

File: examples/meta/generator/parse.py [hunk 2]
```diff
@@ -85,7 +77,8 @@ def addShogunTypes(self, dictionary):
     t_EQUALS = "="
     t_LPAREN = "\("
     t_RPAREN = "\)"
-    t_COMMENT = r"\#.*\n"
```
Comment (karlnapf) [inline examples/meta/generator/parse.py]: comment removed?

**Comment by karlnapf** [comment]
Cool thanks for that. Once we have a json update, I can take over

File: examples/meta/generator/parse.py (hunk 2 above)
Comment (sorig) [inline examples/meta/generator/parse.py]: There was a bug with the parsing error messages not counting comments as a line, so I've moved this rule into one with the side effect of incrementing the line count like the rule for the NEWLINE token.

**Comment by sorig** [comment]
@karlnapf ready for review when travis has finished 

These changes include:
- Support for element access and element assignment for SGVectors and SGMatrices
- Refactor of `translate.py`: dependency extraction and automatic injection of variable storing is now mostly decoupled from translation
- Support for native construction of SGMatrix/SGVector equivalents in target languages. This is implemented for Python. It is an optional feature so doesn't break the other languages.

Does not include:
- Checking that variables being accessed (e.g. `myMatrix[0,2]`) are actually of type SGVector/SGMatrix

**Comment by sorig** [comment]
Here is a [test file](https://gist.github.com/sorig/fa501825e97d33719fbe741d9d988a36) that I used. Would be nice to get a meta example using these features merged, so we can test all the languages.

File: examples/meta/generator/targets/python.json [hunk 3]
```diff
@@ -11,9 +11,33 @@
     "Comment": "#$comment\n",
     "Init": {
         "Construct": "$name = $type($arguments)",
-        "Copy": "$name = $expr"
+        "Copy": "$name = $expr",
+        "BoolVector": "$name = np.zeros( ($arguments), dtype='bool')",
+        "CharVector": "$name = np.zeros( ($arguments), dtype='|S1')",
+        "ByteVector": "$name = np.zeros( ($arguments), dtype='uint8')",
+        "WordVector": "$name = np.zeros( ($arguments), dtype='uint16')",
+        "ShortVector": "$name = np.zeros( ($arguments), dtype=???????)",
```
Comment (sorig) [inline examples/meta/generator/targets/python.json]: I can't import some of these from the Python interface, so can't check what type they should have. Maybe they just need to be removed?

**Comment by karlnapf** [comment]
Let me know when the vector / matrix access stuff is there.
In the test file, it would be good to check both accessing matrix and vector

**Comment by sorig** [comment]
@karlnapf All green. Ready to merge

**Comment by sorig** [comment]
The test file has both a vector and a matrix :)

**Comment by karlnapf** [comment]
Great thanks

@oxphos this will soon allow for vector element access in meta examples, after I added some minor adjustments

**Comment by karlnapf** [comment]
@OXPHOS you should be able to use [something like this](https://gist.github.com/sorig/fa501825e97d33719fbe741d9d988a36) now

Can you try?

**Comment by karlnapf** [comment]
keep in mind to remove old parser output

**Pull request merged**